### ISSUE TITLE
Unshare

### ISSUE THREAD

**Issue by Triantafullenia-Doumani** [pondersource/nc-sciencemesh#114]
## Step 1 : SHARE :green_circle: 
* Marie Shares 'testFile.json' 

```
curl -v -H  'Content-Type:application/json'  -X POST -d '{"md":{"storage_id":"123e4567-e89b-12d3-a456-426655440000","opaque_id":"fileid-marie%2FtestFile.json"},"g":{"grantee":{"type":1,"Id":{"UserId":{"idp":"cernbox.cern.ch","opaque_id":"einstein","type":1}}},"permissions":{"permissions":{"get_path":true,"initiate_file_download":true,"list_container":true,"list_file_versions":true,"stat":true}}}}'  http://marie:radioactivity@localhost:8080/index.php/apps/sciencemesh/~marie/api/ocm/Share
```
## Step 2 : UNSHARE :green_circle: 
* Unshare it

### Curl command
```
curl -v -H  'Content-Type:application/json' -X POST -d  '{"Spec":{"Id":{"opaque_id":"fileid-marie%2FtestFile.json"}}}' http://marie:radioactivity@localhost:8080/index.php/apps/sciencemesh/~marie/api/ocm/Unshare

```
### Result

```
{"Exception":"OCP\\AppFramework\\OCS\\OCSNotFoundException","Message":"Wrong share ID, share doesn't exist","Code":404,
```

### Questions :question: 

* How can we export the share Id from the request the Unshare() receive?
     - Try to get it from the share's path($opaque_id)

**Comment by Triantafullenia-Doumani** [comment]
We will fix that by #117

**Comment by Triantafullenia-Doumani** [comment]
I am trying to use Unshare like this:

### 1. addReceivedShare:
```
curl -v -H  'Content-Type:application/json' -X POST -d '{"md":{"opaque_id":"fileid-einstein%2Ftestfolder3"},"g":{"grantee":{"type":1,"Id":{"UserId":{"idp":"cesnet.cz","opaque_id":"marie","type":1}}}},"provider_domain":"cern.ch","resource_type":"file","provider_id":2,,"protocol":{"name":"webdav","options":{"sharedSecret":"secret","permissions":"webdav-property"}}}' http://marie:radioactivity@localhost:8080/index.php/apps/sciencemesh/~marie/api/ocm/addReceivedShare
```
### 2. Unshare
```
curl -v -H  'Content-Type:application/json' -X POST -d  '{"Spec":{"Id":{"opaque_id":"fileid-einstein%2Ftestfolder3"}}}' http://marie:radioactivity@localhost:8080/index.php/apps/sciencemesh/~marie/api/ocm/GetReceivedShare
```

Response: `< HTTP/1.1 204 No Content`

@br0kk0l1 Am I using the right curl commands?
